File: app/src/main/java/com/test/xyz/demo/domain/model/weather/WeatherRawResponse.java

package com.test.xyz.demo.domain.model.weather;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class WeatherRawResponse {

    @SerializedName("query")
    @Expose
    public Query query;

    public static WeatherRawResponse createWeatherSuccessRawResponse(String temp) {
        WeatherRawResponse weatherRawResponse = new WeatherRawResponse();
        weatherRawResponse.query = new Query();
        weatherRawResponse.query.count = 1;
        weatherRawResponse.query.results = new Results();
        weatherRawResponse.query.results.channel = new Channel();
        weatherRawResponse.query.results.channel.item = new Item();
        weatherRawResponse.query.results.channel.item.condition = new Condition();
        weatherRawResponse.query.results.channel.item.condition.temp = temp;

        return weatherRawResponse;
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/model/weather/WeatherRawResponse.kt
package com.test.xyz.demo.domain.model.weather

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class WeatherRawResponse {

    @SerializedName("query")
    @Expose
    var query: Query? = null

    fun temperature(): String? {
        return query?.results?.channel?.item?.condition?.temp;
    }

    companion object {

        fun createWeatherSuccessRawResponse(temp: String): WeatherRawResponse {
            val weatherRawResponse = WeatherRawResponse()

            weatherRawResponse.query = Query()

            weatherRawResponse.query?.count = 1
            weatherRawResponse.query?.results = Results()
            weatherRawResponse.query?.results?.channel = Channel()
            weatherRawResponse.query?.results?.channel?.item = Item()
            weatherRawResponse.query?.results?.channel?.item?.condition = Condition()
            weatherRawResponse.query?.results?.channel?.item?.condition?.temp = temp

            return weatherRawResponse
        }
    }
}
